evap/static/ts/src/notebook.ts

```diff
@@ -0,0 +1,70 @@
+declare const bootstrap: typeof import("bootstrap");
+import { assert, assertDefinedUnwrap } from "./utils.js";
+import { CSRF_HEADERS } from "./csrf-utils.js";
+
+const NOTEBOOK_LOCALSTORAGE_KEY = "evap_notebook_open";
+
+if (localStorage.getItem(NOTEBOOK_LOCALSTORAGE_KEY) == "true") {
+    assertDefinedUnwrap(document.getElementById("notebook")).classList.add("show");
+    onShowNotebook();
+}
+
+assertDefinedUnwrap(document.getElementById("notebook-save-button")).addEventListener(
+    "click",
+    function (event: MouseEvent): void {
+        event.preventDefault();
+        const form = assertDefinedUnwrap(document.getElementById("notebook-form")) as HTMLFormElement;
+        const data = new FormData(form);
+
+        const target = assertDefinedUnwrap(event.target) as HTMLButtonElement;
+        const default_label = assertDefinedUnwrap(target.getAttribute("value"));
+        const sending_label = assertDefinedUnwrap(target.getAttribute("data-label-sending"));
+        const error_label = assertDefinedUnwrap(target.getAttribute("data-label-error"));
+        const cooldown_label = assertDefinedUnwrap(target.getAttribute("data-label-cooldown"));
+        target.disabled = true;
+        target.setAttribute("value", sending_label);
+
+        fetch(form.action, {
+            body: new URLSearchParams(data as any),
+            headers: CSRF_HEADERS,
+            method: "POST",
+        })
+            .then(response => {
+                assert(response.ok);
+                target.setAttribute("value", cooldown_label);
+                setTimeout(function (): void {
+                    target.setAttribute("value", default_label);
+                    target.disabled = false;
+                }, 2000);
+            })
```

Comment:
@niklasmohrin I find this formatting a bit weird, but it is what prettier currently produces. Is this what we want?

Reply:
Avoiding the discussion about formatting, this should not use `.then` and `.catch` but instead use an async function and `await` the request. It's a bit sus that we are using `fetch` at all, but I suppose it is because we want to reset the UI once the request went through - this "I want my form to tell me when its done" logic could be extracted into its own module, we can also keep it like this for this PR though.